### PR TITLE
Feature/298/treeview search to add blacklist entry

### LINKED ISSUE #298: Create blacklist entry for Treeview search term
# Feature Request
depends on #225

## Description
I some cases, after seeing the result of a search term I want to blacklist these items. 

This makes sense when I search for all `designer.cs` files. First I want to see if my query matches any files and then I want to exclude them. 

This feature would link blacklist and treeview. Maybe we should combine both things into one view then as well.

### PR DESCRIPTION
**Pull request by alschmut**
# Feature/298/treeview search to add blacklist entry

closes #298 
## Description

- Add option button (three dots) in TreeViewSearch to `Hide` or `Exclude` matching nodes
- Disabled `Hide` or `Exclude` buttons, if the pattern is already added with type `Hide` or `Exclude`.
- Removed option to add blacklist entries from inside the blacklistPanel
- Added label in blacklistPanel that refers to add entries from TreeViewSearch or NodeContextMenu
- Use `fa-ban`-icon as symbols for blacklistPanel (instead of `fa-list`) and blacklistType `Exclude` (instead of `fa-times`)
- Add item-counter icon in blacklistPanel header, which counts the number of pattern-entries inside the blacklist (only visible if `list.length > 0`)
- Set short color change animation on item-counter icon when changing the blacklist
- Reorder sidebarPanels -> BlacklistPanel beneath TreeViewSearchPanel
- Clear searchPattern input field, when pattern was blacklisted
- Reorder [`Focus`], `Hide` and `Exclude` buttons in nodeContextMenu and TreeViewSearch options menu
- Remove folderCount
- Add HideCount, ExcludeCount beneath search field with new style

## Definition of Done

A task/pull request will not be considered to be complete until all these items can be checked off.

* [x] **All** requirements mentioned in the issue are implemented
* [x] Does match the Code of Conduct and the Contribution file 
* [x] Task has its own **GitHub issue** (something it is solving)
  * [x] Issue number is **included in the commit messages** for traceability
* [x] **Update the README.md** with any changes/additions made
* [x] **Update the CHANGELOG.md** with any changes/additions made
* [x] **Enough test coverage to ensure that uncovered changes do not break functionality**
* [x] **All tests pass**    
* [x] **Descriptive pull request text**, answering:
  + What problem/issue are you fixing?
  + What does this PR implement and how? 
* [ ] **Assign your PR to someone** for a code review
  + This person _will be contacted **first**_ if a bug is introduced into `master`
* [x] **Manual testing** did not fail

### CODE REVIEW

**Comment by alschmut** [comment]
Just updated the PR with new screenshots showing the new fileCount icons beneath the search field.

File: visualization/app/codeCharta/ui/mapTreeViewSearch/mapTreeViewSearch.component.spec.ts [hunk 1]
```diff
@@ -95,17 +95,84 @@ describe("MapTreeViewSearchController", () => {
...
-        expect(mapTreeViewSearchController.viewModel.fileCount).toBe(0);
-        expect(mapTreeViewSearchController.settingsService.applySettings).toHaveBeenCalled();
+            mapTreeViewSearchController.setSearchedNodePathnames();
+
+            expect(mapTreeViewSearchController.settingsService.settings.searchedNodePaths).toEqual(["/root/a/ab/aba", "/root/a/abc", "/root/a/ab"]);
+            expect(mapTreeViewSearchController.searchedFiles.length).toBe(2);
+            expect(mapTreeViewSearchController.settingsService.applySettings).toHaveBeenCalled();
+        });
+
+        xit("should get correct fileCount, hideCOunt and excludeCount", () => {
```
Comment (mrpatpat) [inline visualization/app/codeCharta/ui/mapTreeViewSearch/mapTreeViewSearch.component.spec.ts]: @alschmut I resolved a bigger merge conflict. I then realized, that this this test fails. I tried running it on the unmerged branch and it still failed. Can you take a look into it ?

Comment (alschmut) [reply visualization/app/codeCharta/ui/mapTreeViewSearch/mapTreeViewSearch.component.spec.ts]: @mrpatpat Alright I gonna talk e a look (:

Comment (alschmut) [reply visualization/app/codeCharta/ui/mapTreeViewSearch/mapTreeViewSearch.component.spec.ts]: @mrpatpat actually it worked for me.. And TravisCI seems to have no problem either

**Pull request merged**